steward/src/cork.rs

```diff
@@ -139,3 +268,14 @@ async fn send_cork(cork: Cork) -> Result<TxResponse, Error> {
     .await
     .map_err(|e| e.into())
 }
+
+async fn contract_call<T: Detokenize>(
+    contract_call: ContractCall<EthSignerMiddleware, T>,
+) -> Result<(), Error> {
+    let gas = contract_call.estimate_gas().await?;
+    let gas_price = CellarGas::get_gas_price().await?;
+    let contract_call = contract_call.gas(gas).gas_price(gas_price);
+    let contract_call = contract_call.send().await?;
+    let _tx_hash = *contract_call;
```

Comment:
I don't think we want to re-assign `contract_call` here, something like `pending_tx` would be more appropriate. When we call `contract_call.send().await?`, we get back a pending transaction. It would be good to log that information so whoever is running the server knows that a transaction was submitted so they can e.g. look it up on Etherscan. The transaction hash is what you get when you dereference the pending transaction (the `*contract_call` bit in the current code).

After you get a pending transaction back, that's still a `Future`. You need to `await` it again for the transaction to fully complete.

Reply:
done!